tests/jballerina-unit-test/src/test/resources/test-src/javainterop/dependently_typed_functions_test.bal

```diff
@@ -241,6 +281,12 @@ function getValue(typedesc<int|float|decimal|string|boolean> td) returns td = @j
     paramTypes: ["io.ballerina.runtime.api.values.BTypedesc"]
 } external;
 
+function getAnnotationValue(typedesc<Foo|Foo2> y) returns map<any> =
```

Comment:
Shall we add a test with return type inferring?
```ballerina
function getAnnotationValue(typedesc<Foo> td = <>) returns td|error =
...
```

Reply:
Updated, please check